addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -706,11 +707,40 @@ export const editorCommands = {
             return selectedNodes.flatMap(node => {
                 let font = closestElement(node, 'font') || closestElement(node, 'span');
                 const children = font && descendants(font);
-                if (font && (font.nodeName === 'FONT' || (font.nodeName === 'SPAN' && font.style[mode]))) {
+                const closestGradient = closestElement(node, '[style*="background-image"]');
+                const isGradient = font && isColorGradient(font.style["background-image"]);
+                const isTextGradient = isGradient && font.classList.contains("text-gradient");
+                if (
+                    font &&
+                    (font.nodeName === "FONT" || (font.nodeName === "SPAN" && font.style[mode])) &&
+                    (isColorGradient(color) || !isGradient)
```

Comment:
This line is very confusing. It looks like we are testing "is it a gradient or is it not a gradient". I suggest renaming `isGradient` to `hasInlineGradient` because it will only be true if the font/span has a gradient. It could very well be `false` while `closestGradientEl` has some element in it, right?